examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+      "    plt.autoscale(autoscale, axis='y')\n",
+      "    plt.plot(t, nengo.spa.similarity(input_net, vocab1), lw=2)\n",
+      "    plt.title(\"Input similarity\")\n",
+      "    plt.xlabel('Time')\n",
+      "    if not autoscale:\n",
+      "        plt.ylim(top=top)\n",
+      "    plt.legend(vocab1.keys, loc='best')\n",
+      "    plt.subplot(1, 2, 2)\n",
+      "    plt.plot(t, nengo.spa.similarity(output_net, vocab2), lw=2)\n",
+      "    plt.title(\"Output similarity\")\n",
+      "    plt.xlabel('Time')\n",
+      "    plt.legend(vocab2.keys, loc='best')\n",
+      "    \n",
+      "plot_similarities(input_net, output_net, vocab, top=1.1)\n"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {
+      "collapsed": true
+     },
+     "source": [
+      "We see that the network input has been constant throughout the simulation (`APPLE=1`). Notice that there might be some small similarity between the pattern `APPLE` and some other patterns. This is because the semantic pointers are not orthogonal and the dot product will amount to some value different from zero. This can be changed by increasing the dimensionality of vectors, yielding more dissimilar representations.\n",
+      "At the output, the similarity of the represented semantic pointer with the semantic pointer `APPLE` increases until it reaches the maximal value (`=1`). This means that the associative memory successfully retrieved a pattern similar to the input pattern. The exponential increase is due to the synaptic filtering in `nengo.Probe`. This is used to show how the input to another group of neurons connected to the output of this particular AM module would look like.\n"
+     ]
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 2: Memory clean-up\n",
+      "\n",
+      "\n",
+      "In the next example we show the ability of the associative memory to retrieve a noisy pattern. \n",
+      "This is a very common use of autoassociative memories, also called \"clean-up\" as the AM removes the noise in the input.\n",
+      "\n",
+      "Now, instead of presenting just one semantic pointer from a vocabulary, we present a combination of several semantic pointers. This is still one single vector, which is mostly similar to one semantic pointer but is also somewhat similar to other semantic pointers. Therefore, as the input is not \"clean\" we can say that we are presenting a noisy pattern.\n",
+      "The effect would have been similar if we had chosen randomly distributed numbers using, for example, `numpy`'s `random.rand` function. To indicate that the output of the AM should be similar to just one stored vector, we set the parameter `wta_output` to `True`. `WTA` is a computational principle called winner-take-all, stating that one, mostly active element should be regarded as the winner among possible, less similar alternatives."
```

Comment:
It should mean "the noisy vector can also be created by adding random numbers to it". 
It's sort of obvious, so I don't know whether this sentence adds anything new. I will probably remove it.